DXMainClient/DXGUI/Generic/LoadingScreen.cs

```diff
@@ -88,9 +94,7 @@ private void Finish()
             ProgramConstants.GAME_VERSION = ClientConfiguration.Instance.ModMode ? 
                 "N/A" : Updater.GameVersion;
 
-            DiscordHandler discordHandler = null;
-            if (!string.IsNullOrEmpty(ClientConfiguration.Instance.DiscordAppId))
-                discordHandler = new DiscordHandler(WindowManager);
```

Comment:
I am not familiar with dependency injection and wonder why removing these lines? What would happen if `ClientConfiguration.Instance.DiscordAppId` is empty?

Reply:
First, dependency injection is a way for the backing host application to determine how to instantiate a given class when specified in the constructor of any given class. So, using `public LoadingScreen(DiscordHandler discordHandler)...` as an example of the `LoadingScreen` constructor, dependency injection will automatically handle the instantiation of the `DiscordHandler` class. That is all configured in the `GameClass` class.

As a result, this line of code is no longer required. We're no longer required to manually instantiate this class.

The dependency injection is set up to ALWAYS create a properly instantiated `DiscordHandler` now. It'll never be null if referenced from a constructor. Therefore, there's no need for null checks. Additionally, the check for `ClientConfiguration.Instance.DiscordAppId` to see if we even can use Discord has been moved into the `DiscordHandler` itself. The `DiscordHandler` knows whether or not it can be used internally and will discard attempts to `UpdatePresence` if it cannot be.